crates/core/src/instance.rs

```diff
@@ -76,6 +76,11 @@ impl Instance {
         scheduler
             .every(10.seconds())
             .run(scheduled_jobs::create_state_dump_callback(context.clone()));
+        scheduler
+            .every(5.seconds())
```

Comment:
Shouldn't there be a direct relationship between ACTION_PRUNE_MS and this number?  I.e. why have it at 3000ms and then only run the task every 5 seconds?

Reply:
Yeah, good point. Also I'm not sure if these numbers are ok anyway. I guess it could make nodes lock up if under stress, if the future that is polling the state for an action doesn't get run before the action result is pruned already - the future would keep waiting for ever.

So I guess we should at least increase `ACTION_PRUNE_MS` to something like 30 or 60 seconds..

Reply:
Shouldn't we just remove the stale data as soon as we don't need it? At soon as the future gets what it needs, shouldn't it just clear the data from the state at that moment? I think we're bound to run into trouble like you mentioned if we don't separate the concerns of actual application state with the concerns of visibility and diagnostics.

Reply:
@maackle good point, but can we guarantee that the data in the state won't be needed after a particular future gets the data? For instance, what if there is a future running in a concurrent thread that is also trying to get the data? There might be a delay between when the future gets the data and it is then stored (if applicable), so if the data needs to be accessed between that time, should that scenario be handled, or prevented from occurring in the first place if it can currently hypothetically occur?

Reply:
Ok, I've added actions and reducers for removing responses from the state separately and dispatch according actions from the futures when they resolve. I have left the general prune in too but set the timings to 60 seconds timeout for responses and a 30 second poll interval for the scheduled job, which I would consider to be safe.

@jamesray1, we don't have any futures share actions. We explicitly make actions unique with a `ProcessUniqueId` in the `ActionWrapper`. So cleaning up the response from the future when it resolves seems like the right way to go here.